### PR TITLE
GraphQL federation example + Upgrade Netbox to v4

### PR DESCRIPTION
**Pull request by Mark90**
# Changes

* Upgraded netbox from v3.4 to v4.0, primarily because they replaced graphene with strawberry
  * Added a Dockerfile for netbox to enable strawberry federation (it's a little bit hacky)
  * Dealt with several breaking changes
* Add apollo-router to docker compose + a one-off service that builds the supergraph from orchestrator and netbox
* Add a strawberry field `NodeBlock { netboxDevice { ... } }` in orchestrator that federates to `DeviceType` in netbox
* Updated `README.md`

## Known issues

* On first run, postgres logs the message `ERROR:  relation "django_content_type" does not exist` a couple of times. This is due to a change in Netbox v3.6 which tries to access this table before it has run its migrations. https://github.com/netbox-community/netbox/pull/13117/files

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -116,7 +148,7 @@ services:
 
   orchestrator:
     &orchestrator
-    image: "ghcr.io/workfloworchestrator/orchestrator-core:latest"
+    image: "ghcr.io/workfloworchestrator/orchestrator-core:edge"  # TODO latest points to 2.2.0 - need 2.3.0rcX with graphql schema fixes
```
Comment (Mark90) [inline docker-compose.yml:151]: Change back to `latest` when 2.3.0 is released

**Comment by Mark90** [comment]
While I've tried to make the netbox code patching robust to change, it could still break. But for purpose of demonstrating GraphQL federation I think it is sufficient.

Probably needs corresponding documentation in workfloworchestrator, but first hoping to hear some opinions on this.

**Review by mrijk — APPROVED** [review]

**Comment by surf-anieuwland** [comment]
Nice PR Mark. It's best if someone with a deeper WFO familiarity also has a look at it, but to me it looks good. I did find the following small things:

* `image: alpine` under the rover-compose service meant that rover-compose did not use the image specified in its build section; removing it resolved it.
* On my Linux machine rover-compose failed because the `docker/federation` directory is not writable. Setting the permissions to 777 resolves it.
* I feel there should be a paragraph mentioning the existance of the federation setup and which URLs they can be accessed under.

**Comment by Mark90** [comment]
> Nice PR Mark. It's best if someone with a deeper WFO familiarity also has a look at it, but to me it looks good. I did find the following small things:
> 
> * `image: alpine` under the rover-compose service meant that rover-compose did not use the image specified in its build section; removing it resolved it.
> * On my Linux machine rover-compose failed because the `docker/federation` directory is not writable. Setting the permissions to 777 resolves it.

Thanks for testing, I've fixed these remarks and will push it soon.

> * I feel there should be a paragraph mentioning the existance of the federation setup and which URLs they can be accessed under.

The readme as it is now briefly discusses the docker compose setup, and then goes into great detail about the orchestrator code structure. I feel like discussing graphql federation here would be distracting, same as we don't discuss things like authentication or celery workers here. 

I'll add documentation in an orchestrator-core PR instead and link it to this one.

**Review by surf-anieuwland — APPROVED** [review]

**Review by hanstrompert — APPROVED** [review]
Nice work including documentation, this should make the use of a federation by others much easier.

**Pull request merged**